contracts/TokenTypeInterface.sol

```diff
@@ -0,0 +1,23 @@
+/*
+ * Hubii Striim
+ *
+ * Compliant with the Hubii Striim specification v0.12.
+ *
+ * Copyright (C) 2017-2018 Hubii AS
+ */
+
+pragma solidity ^0.4.24;
+
+/**
+@title TokenTypeInterface
+@notice A base contract to handle different token types
+*/
+contract TokenTypeInterface {
+    event TokenTransferred(address token, address from, address to, uint256 amount, uint256 id);
```

Comment:
I suggest the latter two parameters be compounded into `amountId`. In controllers of _fungible_ assets then the `receive()` and `send()` functions only will have to require that `amountId > 0`, while controller functions of _non-fungible_ assets only require that `amountId != 0`.

Reply:
Applies the same than answer before. Just if in the future one can collect more than one non-fungible token of the same type.

Reply:
If transferring more than one non-fungible token one would indeed need more than one identifier parameter too.